core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -189,10 +189,15 @@ Queue<HealthCheckContextGroup> contextGroupChain() {
         return contextGroupChain;
     }
 
-    private List<Endpoint> allHealthyEndpoints() {
-        final List<Endpoint> allHealthyEndpoints = new ArrayList<>();
+    @VisibleForTesting
+    List<Endpoint> allHealthyEndpoints() {
         synchronized (contextGroupChain) {
-            final HealthCheckContextGroup newGroup = contextGroupChain.getLast();
+            final HealthCheckContextGroup newGroup = contextGroupChain.pollLast();
+            if (newGroup == null) {
+                return ImmutableList.of();
```

Comment:
Question: Would it make more sense to clear the `contextGroupChain` after all contexts have been released?

So the following line runs when `stopFutures` is completed
https://github.com/line/armeria/blob/88e8d3e6bf7cd204af60166e49c36c8f18e2ea4e/core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java#L296

Reply:
The behavior has no difference. But it makes more sense.